packages/babel-plugin-proposal-decorators/src/transformer-2022-03.ts

```diff
@@ -504,23 +525,62 @@ function transformClass(
     }
 
     if (element.node.decorators && element.node.decorators.length > 0) {
+      switch (element.node.type) {
+        case "ClassProperty":
+          // @ts-expect-error todo: propertyVisitor.ClassProperty should be callable. Improve typings.
+          propertyVisitor.ClassProperty(
+            element as NodePath<t.ClassProperty>,
+            state,
+          );
+          break;
+        case "ClassPrivateProperty":
+          // @ts-expect-error todo: propertyVisitor.ClassPrivateProperty should be callable. Improve typings.
+          propertyVisitor.ClassPrivateProperty(
+            element as NodePath<t.ClassPrivateProperty>,
+            state,
+          );
+          break;
+        case "ClassAccessorProperty":
+          // @ts-expect-error todo: propertyVisitor.ClassAccessorProperty should be callable. Improve typings.
+          propertyVisitor.ClassAccessorProperty(
+            element as NodePath<t.ClassAccessorProperty>,
+            state,
+          );
+          break;
+      }
       hasElementDecorators = true;
     } else if (element.node.type === "ClassAccessorProperty") {
+      // @ts-expect-error todo: propertyVisitor.ClassAccessorProperty should be callable. Improve typings.
+      propertyVisitor.ClassAccessorProperty(
+        element as NodePath<t.ClassAccessorProperty>,
+        state,
+      );
       const { key, value, static: isStatic, computed } = element.node;
 
       const newId = generateClassPrivateUid();
 
-      const valueNode = value ? t.cloneNode(value) : undefined;
-
-      const newField = generateClassProperty(newId, valueNode, isStatic);
+      const newField = generateClassProperty(newId, value, isStatic);
```

Comment:
Here we avoid cloning the accessor property's initializer because
- cloning is unnecessary since the initializer is moved to the proxy setter body
- If the initializer has been transformed by `propertyVisitor.ClassAccessorProperty` (e.g. the case of nested decorator `@noop accessor foo = @dec class {}`), the cloned node will result to a different NodePath, which bypasses the `VISITED` check and gets the inner class `@dec class {}` transformed twice.